source/common/router/config_impl.cc

```diff
@@ -669,6 +669,10 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
         "Specify only one of prefix_rewrite, regex_rewrite or path_rewrite_policy");
   }
 
+  if (!route.route().prefix_rewrite().empty() && path_matcher_ != nullptr) {
```

Comment:
We should be able to directly use `prefix_rewrite_` that has already been initialized, right? This avoids redundant function call to route() and prefix_rewrite().